driver-core/src/main/com/mongodb/connection/InternalStreamConnection.java

```diff
@@ -497,6 +504,11 @@ private ServerAddress getServerAddress() {
         return description.getServerAddress();
     }
 
+    private void updateSessionContext(final SessionContext sessionContext, final ResponseBuffers responseBuffers) {
```

Comment:
Rubber hits road: session context is updated from server response in just this one place